### DISCUSSION TITLE
[Proposal] add `.*` as  an Arithmetic operators

### DISCUSSION THREAD

**Discussion by heartacker** [dotnet/csharplang#7345] [Language Ideas]
just like matlab
https://ww2.mathworks.cn/help/matlab/matlab_prog/matlab-operators-and-special-characters.html

**Comment by TahirAhmadov** [comment]
Can you provide more details? The linked page is in Chinese.

**Reply by tannergooding** [reply]
https://www.mathworks.com/help/matlab/matlab_prog/matlab-operators-and-special-characters.html

They're asking for element-wise multiplication.

**Reply by huoyaoyuan** [reply]
I believe we prefer (extension) methods instead of operators for such cases.

Math-oriented languages often prefer (infix) operators.

**Reply by koszeggy** [reply]
Btw. the regular multiplication operator is already overloaded on [SIMD-accelerated types](https://learn.microsoft.com/en-us/dotnet/standard/simd) such as `Vector<T>`, `Vector4`, `Vector128<T>`, etc. Simply the same `*` operator is [used](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.intrinsics.vector128-1.op_multiply) both for scalar and element-wise multiplication so I don't really see the need for a brand new operator.
